### PR TITLE
Fix Avatar links

### PR DESCRIPTION
**Pull request by nlaz**
#### Related issues

Fixes https://meetup.atlassian.net/browse/WC-28

#### Description
This PR checks whether `href` or `to` props are passed into the component and conditionally returns a `<a>` or `<span>` element. This should fix the problem of not being able to click a `<span>` element in `mup-web`.

#### Prelim
Adding the Prelim tag until I confirm this works on `mup-web`.

### CODE REVIEW

File: src/Avatar.jsx [hunk 1]
```diff
@@ -54,11 +54,19 @@ class Avatar extends React.Component {
...
-				{alt}
-			</span>
-		);
+		if (this.props.to || this.props.href) {
+			return (
+				<a {...allProps}>
+					{alt}
+				</a>
+			);
+		} else {
```
Comment (mmcgahan) [inline src/Avatar.jsx]: no need for `else` if the `if` returns

Comment (nlaz) [inline src/Avatar.jsx]: good call 👍

**Comment by nlaz** [comment]
Verified this change works in `mup-web`. Removed the `Prelim` tag.

File: src/avatar.test.js [hunk 2]
```diff
@@ -37,4 +37,23 @@ describe('Avatar', function() {
...
+
+		expect(avatarComponent.tagName.toLowerCase()).toBe('a');
+	});
+	it('renders a link element if `to` passed in', () => {
+		const avatar = TestUtils.renderIntoDocument(<Avatar className='test' to='/' />);
+		const avatarComponent = TestUtils.findRenderedDOMComponentWithClass(avatar, 'test');
+
+		expect(avatarComponent.tagName.toLowerCase()).toBe('a');
+	});
+	it('renders a span element if `href` passed in', () => {
```
Comment (mmcgahan) [inline src/avatar.test.js]: I think this description should be `renders a span element if no 'href' or 'to' passed in`

File: src/Avatar.jsx [hunk 3]
```diff
@@ -54,6 +54,14 @@ class Avatar extends React.Component {
 			...other
 		};
 
+		if (this.props.to || this.props.href) {
+			return (
+				<a {...allProps}>
+					{alt}
+				</a>
+			);
```
Comment (mmcgahan) [inline src/Avatar.jsx]: trivial, but this is a lot of line breaks for a small element - unless you expect a lot of embellishment of the `<a>`, I recommend keeping it tight to keep the overall component more readable

```jsx
return <a {...allProps}>{alt}</a>
```

Comment (nlaz) [inline src/Avatar.jsx]: I also like it concise like this

**Pull request merged**